### PR TITLE
Rename Mock OT impls to Ideal OT

### PR DESCRIPTION
**Pull request by sinui0**
It's more conventional to call these "ideal" implementations of each functionality. This PR only updates OT, we'll want to rename stuff in other crates as well.

### CODE REVIEW

File: ot/mpz-ot/src/ideal/shared.rs [hunk 1]
```diff
@@ -32,21 +32,21 @@ pub fn mock_ot_shared_pair() -> (MockSharedOTSender, MockSharedOTReceiver) {
 /// A mock oblivious transfer sender.
```
Comment (th4s) [inline ot/mpz-ot/src/ideal/shared.rs]: Should this comment also be adapted?

File: ot/mpz-ot/src/ideal/shared.rs [hunk 2]
```diff
@@ -69,13 +69,13 @@ impl<T: Clone + std::fmt::Debug + Send + Sync + 'static> CommittedOTSenderShared
 /// A mock oblivious transfer receiver.
```
Comment (th4s) [inline ot/mpz-ot/src/ideal/shared.rs]: same

File: ot/mpz-ot/src/ideal/mod.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+//! Mock implementations of the OT protocols.
```
Comment (th4s) [inline ot/mpz-ot/src/ideal/mod.rs]: Should this comment be adapted?

**Review by th4s — APPROVED** [review]
:+1:

**Review by themighty1 — APPROVED** [review]

**Pull request merged**